source/testing/testing-components.md

```diff
@@ -133,25 +181,26 @@ We recommend using native DOM events wrapped inside the run loop or the [`ember-
 Using jQuery to simulate user click events might lead to unexpected test results as the action can potentially be called twice.
 
 ```tests/integration/components/magic-title-test.js
-import { moduleForComponent, test } from 'ember-qunit';
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
 import hbs from 'htmlbars-inline-precompile';
-import { run } from '@ember/runloop';
 
-moduleForComponent('magic-title', 'Integration | Component | magic title', {
-  integration: true
-});
+module('Integration | Component | magic title', function(hooks) {
+  setupRenderingTest(hooks);
 
-test('should update title on button click', function(assert) {
-  assert.expect(2);
+  test('should update title on button click', async function(assert) {
+    assert.expect(2);
 
-  this.render(hbs`{{magic-title}}`);
+    await render(hbs`{{magic-title}}`);
 
-  assert.equal(this.$('h2').text(), 'Hello World', 'initial text is hello world');
+    assert.equal(this.element.querySelector('h2').textContent.trim(), 'Hello World', 'initial text is hello world');
 
-  //Click on the button
-  run(() => document.querySelector('.title-button').click());
+    //Click on the button
+    run(() => this.element.querySelector('.title-button').click());
```

Comment:
this should use `await click('.title-button')` instead

Reply:
The helpers for DOM interaction are just introduced later on in this section, so I believe it would make sense to leave it at that for now and update that in another PR for restructuring / improving the testing guides in general

Reply:
I disagree. The pattern above is obsolete and should not be used anymore.

Reply:
I strongly agree with @Turbo87. Invoking `.click()` on an element (and relying on synchronous updating) is fundamentally bad (this was one of the main factors in deciding to ship interaction helpers)...

Reply:
> The helpers for DOM interaction are just introduced later on in this section, so I believe it would make sense to leave it at that for now and update that in another PR for restructuring / improving the testing guides in general

Should the intro to the DOM interaction helpers be moved to the very beginning of the component testing section then in this PR already so we can refer to that in the subsequent code examples?

Reply:
in the current structure acceptance/application tests are introduced first which also use these helpers, right? shouldn't they be introduced in that chapter then?